src/api/nft.ts

```diff
@@ -1,15 +1,69 @@
 import { NftMetadata, NftTokenType, TokenUri } from '../types/types';
-import { RawBaseNft, RawNft } from '../internal/raw-interfaces';
+import {
+  RawBaseNft,
+  RawBaseNftContract,
+  RawNft,
+  RawNftContract
+} from '../internal/raw-interfaces';
 import { BigNumber } from 'ethers';
 
+export interface NftContractMetadata {
+  /** The name of the contract. */
+  name: string;
```

Comment:
These fields should be marked optional since docs state that the fields are only set if the contract implements them.

`tokenType` should always be defined for the end-developer, since it should default to `UNKNOWN` if not specified by the backend